libyul/backends/wasm/EVMToEwasmTranslator.cpp

```diff
@@ -137,7 +137,7 @@ void EVMToEwasmTranslator::parsePolyfill()
 			string(solidity::yul::wasm::polyfill::Logical) +
 			string(solidity::yul::wasm::polyfill::Memory) +
 		"}", "");
-	m_polyfill = Parser(errorReporter, WasmDialect::instance()).parse(charStream);
+	m_polyfill = Parser(errorReporter, WasmDialect::instance(), langutil::SourceLocation()).parse(charStream);
```

Comment:
Hmm, why do we have `Parser` with a default value (without the explicit arg) if it causes crashes?

Reply:
It's a bit more complicated. After #11998 we can discern between the native locations and those grabbed from annotations. In normal yul compilation it does not cause a crash even though the the original source is not available. So why does it crash here?

I think that the problem is actually that the `nativeLocation`s in the generated AST are wrong. `EVMToEwasmTranslator` [copies the whole debug info](https://github.com/ethereum/solidity/blob/v0.8.10/libyul/backends/wasm/EVMToEwasmTranslator.cpp#L96), when it translates the AST. This means that the new AST gets the native location pointing at the original assembly input file rather than the translated source it's supposed to point at.

The problem is that the translated assembly does not actually exist until we generate it from the AST. At the point where the debug info is being copied, we only have the (incomplete) AST and not source we could point at. I think the only thing we can do is to use empty `nativeLocation` in the translated AST. Not having native location means that we can't use it in errors reported from the analysis of that AST. But I don't think we even analyze it - it's generated by the compiler after all so it's not supposed to have errors - those should have been caught during parsing of the original assembly input. It's very likely that the only code that actually tries to access the native location is the `AsmPrinter` (which is where this crash happens).

The location override is another way to get rid of native locations from the AST but a very heavy-handed one - it also discards the origin location. Currently it does not matter much, because I don't think we make use of it when generating ewasm. It will matter once we decide we want to include debug info in ewasm though.

Overall, I think this is an acceptable workaround as long as work on ewasm is stalled. I'd just add a comment explaining that this is just a workaround and that it effectively strips both `originLocation` and `nativeLocation` from the AST even though only `nativeLocation` really has to be discarded.